ci_framework/playbooks/06-deploy-edpm.yml

```diff
@@ -49,10 +49,71 @@
             name: libvirt_manager
             tasks_from: deploy_edpm_compute.yml
 
+        - name: Disable discover_hosts when deploying HCI phase1
+          when: cifmw_edpm_deploy_hci | default(false) | bool
+          ansible.builtin.set_fact:
+            cifmw_edpm_deploy_skip_nova_discover_hosts: true
+
         - name: Deploy EDPM
           ansible.builtin.import_role:
             name: edpm_deploy
 
+- name: Clear edpm hosts facts to force refreshing them
+  hosts: edpm
+  tasks:
+    - name: Clear edpm hosts facts
+      when: cifmw_edpm_deploy_hci | default(false) | bool
+      ansible.builtin.meta: clear_facts
```

Comment:
clear_facts for edpm nodes. After deploy network services for each EDPM node, we need to gather all facts again. With gathering=smart we don't have facts refreshing, unless we clear them. Another option would be to change default gathering configuration.

Reply:
interesting, i have not come across this before https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html

apparently it will clear facts for those hosts that this play is running against, which is fine since it is the same hosts that you will re-run config/deploy with

Reply:
Yes, that is needed to gather all networking facts again, since in edpm_deploy step, new interfaces are added for network isolantion, and ceph.yml script need to know which IP was configured there.
This is needed due to https://github.com/openstack-k8s-operators/ci-framework/blob/f7038a6495825e0ff1a4fb5e2c64d5701ab9a7a8/ansible.cfg#L9
which will not update facts again since edpm node's facts already exist

Reply:
thanks going to mark resolved